backend/prompt_studio/prompt_studio_index_manager/models.py

```diff
@@ -90,31 +91,40 @@ class Meta:
         ]
 
 
+def delete_from_vector_db(index_ids_history, vector_db_instance_id):
+    org_schema = connection.tenant.schema_name
+    util = PromptIdeBaseTool(log_level=LogLevel.INFO, org_id=org_schema)
+    vector_db = VectorDB(
+        tool=util,
+        adapter_instance_id=vector_db_instance_id,
+    )
+    for index_id in index_ids_history:
+        logger.debug(f"Deleting from VectorDB - index id: {index_id}")
+        try:
+            vector_db.delete(ref_doc_id=index_id)
+        except Exception as e:
+            # Log error and continue with the next index id
+            logger.error(f"Error deleting index: {index_id} - {e}")
+
+
 # Function will be executed every time an instance of IndexManager is deleted.
-@receiver(post_delete, sender=IndexManager)
+@receiver(pre_delete, sender=IndexManager)
 def perform_vector_db_cleanup(sender, instance, **kwargs):
     """Signal to perform vector db cleanup."""
     logger.info("Performing vector db cleanup")
-    logger.debug("Document tool id: %s", instance.document_manager.tool_id)
+    logger.debug(f"Document tool id: {instance.document_manager.tool_id}")
     try:
         # Get the index_ids_history to clean up from the vector db
         index_ids_history = json.loads(instance.index_ids_history)
         vector_db_instance_id = str(instance.profile_manager.vector_store.id)
-        # Generate a run_id
-        org_schema = connection.tenant.schema_name
-        util = PromptIdeBaseTool(log_level=LogLevel.INFO, org_id=org_schema)
-        vector_db = VectorDB(
-            tool=util,
-            adapter_instance_id=vector_db_instance_id,
-        )
-        for index_id in index_ids_history:
-            logger.debug(f"Deleting from VectorDB - index id: {index_id}")
-            try:
-                vector_db.delete(ref_doc_id=index_id)
-            except Exception as e:
-                # Log error and continue with the next index id
-                logger.error(f"Error deleting index: {index_id} - {e}")
-    # Not raising any exception.
-    # Cleanup should not fail the deletion of the index manager.
+        delete_from_vector_db(index_ids_history, vector_db_instance_id)
     except SdkError as e:
         logger.error(f"Error while performing vector db cleanup: {e}")
+    except ObjectDoesNotExist:
+        logger.error(
+            f"No ProfileManager found for tool_id {instance.document_manager.tool_id}"
+        )
+    except AttributeError:
+        logger.error("ProfileManager or vector_store is None")
+    except Exception as e:
+        logger.error(f"An unexpected error occurred: {e}")
```

Comment:
```suggestion
    except Exception as e:
        logger.error(f"Error during vector DB {vector_db_instance_id} cleanup for deleted document in prompt studio tool {instance.document_manager.tool_id}: {e}") from e
```
Let's retain this alone, make sure to lint the line length

Reply:
updated

Reply:
NIT: Will be good to add context on which vector DB caused the error as well to have context rich logging